benchexec/tools/ultimate.py

```diff
@@ -333,10 +333,19 @@ def program_files(self, executable):
 
     def determine_result(self, run):
         if any(arg for arg in run.cmdline if "--spec" == arg or ".prp" in arg):
-            return self._determine_result_with_property_file(run)
+            return self._determine_result_with_property_file(self, run)
```

Comment:
A method call with `self.f(self)` is just weird. Same for declaring `@staticmethod` with a `self` parameter. While it technically works, I see no reason to use it. A method intended for overwriting should simply not be declared `@staticmethod`.

Note that this and the whole other changes around the methods here would go away if the parsing logic for TestGen is simply inlined.